packages/core/src/metadata/ng_module.ts

```diff
@@ -136,7 +136,7 @@ export interface NgModule {
    * ```
    *
    */
-  imports?: Array<Type<any>|ModuleWithProviders<{}>|any[]>;
+  imports?: Array<Type<any>|ModuleWithProviders<{}>|ReadonlyArray<any>>;
```

Comment:
Should this not be:
```suggestion
  imports?: Array<Type<any>|ModuleWithProviders<{}>|ReadonlyArray<Type<any>>>;
```
in which case I think we can just do:
```suggestion
  imports?: ReadonlyArray<Type<any>|ModuleWithProviders<{}>;
```
as all `Array`s are implicitly `ReadonlyArray`s.

Reply:
Can `Type<any>` represent an array of providers? Previously, it was valid to import an array of providers, like this:
```
const providers = [/*...*/];
import: [providers]
```

IMO keeping the type as-is can be more descriptive about what the `import` accepts.